### DISCUSSION TITLE
Strict Mode error when attempting to programmatically delete a relation field element

### DISCUSSION THREAD

**Discussion by onsombal** [strapi/strapi#8556] [Legacy Topics]
I’m wondering if anyone has encountered the issue I’m having with attempting to programmatically delete an element from a relation field. The relation field is an array. I can easily add element objects to the relation, but when I attempt to delete an element I get a Javascript Strict Mode error:

- TypeError – Can’t delete non-configurable array element (_message returned by FireFox_)
- TypeError – Cannot delete property ‘1’ of [object Array] (_message returned by Chrome_)

Both of the error messages refer to the same Strict Mode Error relating to not allowing the shortening of a non-configurable array.

Clearly, there is a way to do this since I can delete a relation element from the Strapi UI panel. My code follows:
```
   // get index of object
      let updateValue = cartItems
        .map(function (item) {
          return item.id;
        })
        .indexOf(currentUser);
        console.log("updateValue " + updateValue);  // this returns 1 as the index position of the target element in a two-item array.
      // remove object
      cartItems.splice(updateValue, 1);
```
The code above finds the index of the target element, but the Array.splice() method fails, reporting the non-configurable array error.

Thoughts?

Thanks… Jim (edited)

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question, we are currently planning to decomission our GitHub Discussions in early 2021 and we don't want to lose the great questions asked by our community.

Questions should be directed to [our forum](https://forum.strapi.io/). Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).

**Reply by onsombal** [reply]
@derrickmehaffy -- Yes, I saw that notice after I posted. I replicated this post there as well.

Thank you!
